activerecord/lib/active_record/railties/query_comments.rb

```diff
@@ -0,0 +1,40 @@
+module ActiveRecord
+  module Railties # :nodoc:
+    module QueryComments #:nodoc:
+      extend ActiveSupport::Concern
+
+      included do
+        # TODO make this optional based on config, default enabled
+        class_eval do
+          around_action :record_query_comment
+        end
+        ActiveRecord::ConnectionAdapters::AbstractAdapter::Context.include(ControllerContext)
+        ActiveRecord::ConnectionAdapters::AbstractAdapter::Context.components = [:application, :controller, :action]
```

Comment:
This feels like it shouldn't be in the included block, since it doesn't act on the class being included on. Maybe this should be in the `on_load` from the railtie.

Reply:
Yeah makes sense - may lose some proximity context (easy to see that we're adding components based on the methods defined in the module here), but that doesn't 100% hold water as we're not adding _all_ of them. 👍